src/server/http/__tests__/xsrf.js

```diff
@@ -71,6 +85,24 @@ describe('xsrf request filter', function () {
         if (method === 'HEAD') expect(resp.payload).to.be.empty();
         else expect(resp.payload).to.be('ok');
       });
+
+      it('accepts requests with any content-type header', async function () {
+        const resp = await inject(kbnServer, {
+          url: '/xsrf/test/route',
+          method: method,
+          headers: {
+            [contentTypeHeader]: 'anything',
+          },
+        });
+
+        expect(resp.statusCode).to.be(200);
+
+        if (method === 'HEAD') {
```

Comment:
Not a fan of `if`s within tests. As we special-case `HEAD` in all the test within this `for` I think we should split out separate tests for `GET` and `HEAD` instead.